src/main/java/org/apache/sysds/runtime/matrix/data/LibCommonsMath.java

```diff
@@ -295,11 +281,283 @@ private static MatrixBlock computeMatrixInverse(Array2DRowRealMatrix in) {
 	 * @return matrix block
 	 */
 	private static MatrixBlock computeCholesky(Array2DRowRealMatrix in) {
-		if ( !in.isSquare() )
-			throw new DMLRuntimeException("Input to cholesky() must be square matrix -- given: a " + in.getRowDimension() + "x" + in.getColumnDimension() + " matrix.");
+		if(!in.isSquare())
+			throw new DMLRuntimeException("Input to cholesky() must be square matrix -- given: a "
+				+ in.getRowDimension() + "x" + in.getColumnDimension() + " matrix.");
 		CholeskyDecomposition cholesky = new CholeskyDecomposition(in, RELATIVE_SYMMETRY_THRESHOLD,
 			CholeskyDecomposition.DEFAULT_ABSOLUTE_POSITIVITY_THRESHOLD);
 		RealMatrix rmL = cholesky.getL();
 		return DataConverter.convertToMatrixBlock(rmL.getData());
 	}
+
+	/**
+	 * Function to perform the Lanczos algorithm and then computes the Eigendecomposition.
+	 * Caution: Lanczos is not numerically stable (see https://en.wikipedia.org/wiki/Lanczos_algorithm)
+	 * Input must be a symmetric (and square) matrix.
+	 *
+	 * @param in matrix object
+	 * @return array of matrix blocks
+	 */
+	private static MatrixBlock[] computeEigenLanczos(MatrixBlock in) {
+		if(in.getNumRows() != in.getNumColumns()) {
+			throw new DMLRuntimeException(
+				"Lanczos algorithm and Eigen Decomposition can only be done on a square matrix. "
+					+ "Input matrix is rectangular (rows=" + in.getNumRows() + ", cols=" + in.getNumColumns() + ")");
+		}
+		if(!isSym(in)) {
```

Comment:
this isSym check is taking much of your time, 
In the computeEigen we have skipped this check, i think it is fair to do the same here.